src/App.vue

```diff
@@ -324,23 +389,119 @@ export default {
+    },
+
+    // FUTURE: rebase this to latest code from bcrs-business-create-ui
+    // FUTURE: maybe this should be in a mixin
+    /** Returns True if NR is consumable. */
+    isNrConsumable (data: any): boolean {
+      const nr = this.isNRConsumable(data)
+
+      if (nr.expired) {
+        // NR has expired
+        this.nameRequestInvalidType = NameRequestStates.EXPIRED
+        return false
+      }
+      if (!nr.approved) {
+        // NR has not been approved
+        this.nameRequestInvalidType = NameRequestStates.NOT_APPROVED
+        return false
+      }
+      if (!nr.isConsumable) {
+        // NR has already been consumed
+        this.nameRequestInvalidType = NameRequestStates.CONSUMED
+        return false
+      }
+      if (nr.approved && data.consentFlag === false) {
+        // NR is awaiting consent
+        this.nameRequestInvalidType = NameRequestStates.NEED_CONSENT
+        return false
+      }
+
+      return nr.isConsumable
+    },
+
     /** Gets tasks list from Legal API. */
     getTasks (): Promise<any> {
-      const url = this.businessId + '/tasks'
+      const id = this.businessId || this.nrNumber
+      const url = `businesses/${id}/tasks`
       return axios.get(url)
     },
 
     storeTasks (response: any): void {
-      if (response && response.data && response.data.tasks) {
-        this.setTasks(response.data.tasks)
-      } else {
-        throw new Error('Invalid tasks')
+      if (this.businessId) {
+        if (response && response.data && response.data.tasks) {
+          this.setTasks(response.data.tasks)
+        } else {
+          throw new Error('Invalid tasks')
+        }
+      }
+      // FUTURE: update this when API returns New Incorporation task (#3102)
+      if (this.nrNumber) {
+        if (response && response.data && response.data.tasks) {
+          // if we have existing tasks, use them
+          if (response.data.tasks.length > 0) {
+            this.setTasks(response.data.tasks)
+            this.setEntityStatus(EntityStatus.INCORPORATION_APPLICATION)
+          } else {
+            // otherwise create a New Incorporation task
+            response.data.tasks.push({
+              enabled: true,
+              order: 1,
+              task: {
+                todo: {
+                  nameRequest: this.nameRequest,
+                  header: {
+                    name: FilingTypes.NAME_REQUEST,
+                    status: FilingStatus.NEW
+                  }
+                }
+              }
+            })
+            this.setTasks(response.data.tasks)
+            this.setEntityStatus(EntityStatus.NAME_REQUEST)
+          }
+        } else {
+          throw new Error('Invalid tasks')
+        }
       }
     },
 
     /** Gets filings list from Legal API. */
     getFilings (): Promise<any> {
-      const url = this.businessId + '/filings'
+      const id = this.businessId || this.nrNumber
```

Comment:
Interesting change! It may be uncharted waters but what happens after a user completes their incorporation application. At what point do we stop using the NR for the identifier and start using the BusinessID?

Reply:
Um... (thinking)

We will do whatever Scott designs UX-wise, and this also depends on what BE processing needs to happen for a draft incorporation to become a business, but it would be nice if we could redirect the user from `...cooperatives/NRxxxxxxx` to `...cooperatives/CPxxxxxxx`.